File: api/laravel10.x/tests/unit/Gateway/Repositories/UserRepositoryImplTest.php
<?php

namespace Tests\Unit\Gateway\Repositories;

use App\Domain\Entities\User;
use App\Gateway\Repositories\Models\EloquentUser;
use App\Gateway\Repositories\UserRepositoryImpl;
use Illuminate\Foundation\Testing\RefreshDatabase;
use Tests\TestCase;

class UserRepositoryImplTest extends TestCase
{
  use RefreshDatabase;

  public function test_ユーザをDBに追加できること()
  {
    // Arrange
    $exceptedUser = new User('John Doe', 'john@example.com');

    // Act
    $repository = new UserRepositoryImpl();

    $user = new User('John Doe', 'john@example.com');
    $actualUser = $repository->create($user);

    // Assert
    $this->assertDatabaseHas('users', [
      'name' => $exceptedUser->getName(),
      'email' => $exceptedUser->getEmail(),
    ]);
    $this->assertEquals($actualUser, $exceptedUser);
  }

  public function test_DBにあるユーザを全部取得できること()
  {
    // Arrange
    $expectedUsers = [
      new User('John Doe', 'john@example.com'),
      new User('Jane Doe', 'jane@example.com')
    ];
    // Database準備
    EloquentUser::create([
      'name' => 'John Doe',
      'email' => 'john@example.com',
    ]);
    EloquentUser::create([
      'name' => 'Jane Doe',
      'email' => 'jane@example.com',
    ]);

    // Act
    $repository = new UserRepositoryImpl();
    $actualUsers = $repository->getAll();

    // Assert
    $this->assertEquals($actualUsers, $expectedUsers);
  }
}

File: api/springboot/src/test/java/com/example/demo/adapters/gateway/repositories/UserRepositoryImplTest.java
package com.example.demo.adapters.gateway.repositories;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;

import com.example.demo.domain.model.User;
import com.example.demo.infrastructure.db.jpa.entities.UserJpaEntity;
import com.example.demo.infrastructure.db.jpa.repositories.UserJpaRepository;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertAll;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

@DataJpaTest
class UserRepositoryImplTest {

  @Autowired
  private UserJpaRepository userJpaRepository;

  @Test
  void test_ユーザ情報を正しくデータベースに保存できること() {
    // Arrange
    UserRepositoryImpl userRepositoryImpl = new UserRepositoryImpl(userJpaRepository);
    User expected = new User("John Doe", "john@example.com");

    // Act
    User actual = userRepositoryImpl.save(expected);

    // Assert
    assertNotNull(actual);
    assertEquals(expected.getName(), actual.getName());
    assertEquals(expected.getEmail(), actual.getEmail());

    // データベースにデータが確かに保存されたかを確認
    List<UserJpaEntity> actualFromDatabase = userJpaRepository.findAll();
    assertNotNull(actualFromDatabase);
    assertEquals(1, actualFromDatabase.size());
    assertAll("UserJpaEntity fields",
        () -> assertEquals(expected.getName(), actualFromDatabase.get(0).getName()),
        () -> assertEquals(expected.getEmail(), actualFromDatabase.get(0).getEmail()));
  }
}
